lib/google/gax/settings.rb

```diff
@@ -62,20 +62,22 @@ class CallSettings
       #   performed.
       # @param bundle_descriptor [BundleDescriptor] indicates the structure of
       #   the bundle. If nil, bundling is not performed.
+      # @param metadata [Hash] the request header params.
       # @param kwargs [Hash]
-      #   Additional keyword argments to be passed to the API call.
+      #   Deprecated, if set this will be merged with the metadata field.
       # @param errors [Array<Exception>]
       #   Configures the exceptions to wrap with GaxError.
       def initialize(timeout: 30, retry_options: nil, page_descriptor: nil,
                      page_token: nil, bundler: nil, bundle_descriptor: nil,
-                     kwargs: {}, errors: [])
+                     metadata: {}, kwargs: {}, errors: [])
         @timeout = timeout
         @retry_options = retry_options
         @page_descriptor = page_descriptor
         @page_token = page_token
         @bundler = bundler
         @bundle_descriptor = bundle_descriptor
-        @kwargs = kwargs
+        @metadata = metadata
+        @metadata.merge!(kwargs) if kwargs && metadata
```

Comment:
Might be good to add a deprecation warning to this.

Reply:
I assume kwargs will in a manner with no time-frame of removal and removed at next major update, correct?

Reply:
Yup, is there a better way to add that note than what I've got?

Reply:
You could use the [Kernel#warn](http://ruby-doc.org/core-2.2.0/Kernel.html#method-i-warn) method to simply emit a warning. We have done that before but I don't know if there is a more idiomatic way than that.